[CI] TransportClusterStateActionTests testGetClusterStateWithDefaultProjectOnly failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #10541 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/ommwxkp6pcxam)
- [elasticsearch-periodic-platform-support #10541 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/wbdjsa33zaktq)
- [elasticsearch-periodic-platform-support #10541 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/6wzre3ohqafpe)
- [elasticsearch-periodic #10551 / openjdk21_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/ble53cml3r4d4)
- [elasticsearch-pull-request #91064 / part-1-fips](https://gradle-enterprise.elastic.co/s/ty34cdbufteao)
- [elasticsearch-intake #28221 / part1](https://gradle-enterprise.elastic.co/s/qlp562f4xfnom)
- [elasticsearch-intake #28220 / part1](https://gradle-enterprise.elastic.co/s/ftzozogglwbye)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.action.admin.cluster.state.TransportClusterStateActionTests.testGetClusterStateWithDefaultProjectOnly" -Dtests.seed=7C98BA2664E66BE8 -Dtests.locale=it-VA -Dtests.timezone=Europe/Samara -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.action.admin.cluster.state.TransportClusterStateActionTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testGetClusterStateWithDefaultProjectOnly'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a map with size <2>
     but: map size was <1>
```

**Issue Reasons:**
- [main] 7 failures in test testGetClusterStateWithDefaultProjectOnly (2.5% fail rate in 282 executions)
- [main] 2 failures in step part1 (5.3% fail rate in 38 executions)
- [main] 2 failures in pipeline elasticsearch-intake (5.3% fail rate in 38 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -186,9 +186,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.ClassificationIT
   method: testWithCustomFeatureProcessors
   issue: https://github.com/elastic/elasticsearch/issues/134001
-- class: org.elasticsearch.action.admin.cluster.state.TransportClusterStateActionTests
-  method: testGetClusterStateWithDefaultProjectOnly
-  issue: https://github.com/elastic/elasticsearch/issues/134450
 - class: org.elasticsearch.xpack.esql.plugin.CanMatchIT
   method: testAliasFilters
   issue: https://github.com/elastic/elasticsearch/issues/134512

--- a/server/src/test/java/org/elasticsearch/action/admin/cluster/state/TransportClusterStateActionTests.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/cluster/state/TransportClusterStateActionTests.java
@@ -101,7 +101,7 @@ public class TransportClusterStateActionTests extends ESTestCase {
 
         final ProjectId projectId = Metadata.DEFAULT_PROJECT_ID;
         final ProjectMetadata.Builder project = projectBuilder(projectId, indexNames);
-        final ClusterState state = buildClusterState(project);
+        final ClusterState state = buildClusterState(false, project);
 
         final ClusterStateResponse response = executeAction(projectResolver, request, state);
 
@@ -229,7 +229,7 @@ public class TransportClusterStateActionTests extends ESTestCase {
         } else {
             assertThat(routingTables.get(projectId).indicesRouting(), anEmptyMap());
         }
-        if (request.customs()) {
+        if (request.customs() && multiprojectEnabled) {
             ProjectStateRegistry projectStateRegistry = ProjectStateRegistry.get(response.getState());
             assertThat(projectStateRegistry.size(), equalTo(1));
             Settings projectSettings = projectStateRegistry.getProjectSettings(projectId);
@@ -280,6 +280,11 @@ public class TransportClusterStateActionTests extends ESTestCase {
     }
 
     private static ClusterState buildClusterState(ProjectMetadata.Builder... projects) {
+        return buildClusterState(true, projects);
+    }
+
+    private static ClusterState buildClusterState(boolean multiprojectEnabled, ProjectMetadata.Builder... projects) {
+        assert multiprojectEnabled || projects.length == 1;
         final Metadata.Builder metadataBuilder = Metadata.builder();
         metadataBuilder.put(
             ReservedStateMetadata.builder("file_settings")
@@ -292,14 +297,16 @@ public class TransportClusterStateActionTests extends ESTestCase {
 
         ClusterState.Builder csBuilder = ClusterState.builder(new ClusterName(randomAlphaOfLengthBetween(4, 12)));
         ProjectStateRegistry.Builder psBuilder = ProjectStateRegistry.builder();
-        for (ProjectMetadata.Builder project : projects) {
-            psBuilder.putReservedStateMetadata(
-                project.getId(),
-                ReservedStateMetadata.builder("file_settings")
-                    .version(43L)
-                    .putHandler(new ReservedStateHandlerMetadata("project_settings", Set.of("setting_1")))
-                    .build()
-            ).putProjectSettings(project.getId(), Settings.builder().put("setting_1", randomIdentifier()).build());
+        if (multiprojectEnabled) {
+            for (ProjectMetadata.Builder project : projects) {
+                psBuilder.putReservedStateMetadata(
+                    project.getId(),
+                    ReservedStateMetadata.builder("file_settings")
+                        .version(43L)
+                        .putHandler(new ReservedStateHandlerMetadata("project_settings", Set.of("setting_1")))
+                        .build()
+                ).putProjectSettings(project.getId(), Settings.builder().put("setting_1", randomIdentifier()).build());
+            }
         }
         return csBuilder.metadata(metadata)
             .routingTable(GlobalRoutingTableTestHelper.buildRoutingTable(metadata, RoutingTable.Builder::addAsNew))

--- a/server/src/test/java/org/elasticsearch/action/admin/cluster/state/TransportClusterStateActionTests.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/cluster/state/TransportClusterStateActionTests.java
@@ -106,7 +106,7 @@ public class TransportClusterStateActionTests extends ESTestCase {
         final ClusterStateResponse response = executeAction(projectResolver, request, state);
 
         assertThat(response.getClusterName(), equalTo(state.getClusterName()));
-        assertSingleProjectResponse(request, response, projectId, expectedIndices);
+        assertSingleProjectResponse(request, response, projectId, expectedIndices, false);
     }
 
     public void testGetClusterStateForOneProjectOfMany() throws Exception {
@@ -128,7 +128,7 @@ public class TransportClusterStateActionTests extends ESTestCase {
         final ClusterStateResponse response = executeAction(projectResolver, request, state);
 
         assertThat(response.getClusterName(), equalTo(state.getClusterName()));
-        assertSingleProjectResponse(request, response, projectId, expectedIndices);
+        assertSingleProjectResponse(request, response, projectId, expectedIndices, true);
     }
 
     public void testGetClusterStateForManyProjects() throws Exception {
@@ -186,7 +186,8 @@ public class TransportClusterStateActionTests extends ESTestCase {
         ClusterStateRequest request,
         ClusterStateResponse response,
         ProjectId projectId,
-        String[] expectedIndices
+        String[] expectedIndices,
+        boolean multiprojectEnabled
     ) {
         final Metadata metadata = response.getState().metadata();
         assertThat(metadata.projects().keySet(), contains(projectId));
@@ -200,13 +201,21 @@ public class TransportClusterStateActionTests extends ESTestCase {
                 assertNotNull(fileSettings);
                 assertThat(fileSettings.version(), equalTo(43L));
                 Map<String, ReservedStateHandlerMetadata> handlers = fileSettings.handlers();
-                assertThat(handlers, aMapWithSize(2));
+                if (multiprojectEnabled) {
+                    assertThat(handlers, aMapWithSize(2));
+                } else {
+                    assertThat(handlers, aMapWithSize(1));
+                }
                 ReservedStateHandlerMetadata clusterSettingsHandler = handlers.get("cluster_settings");
                 assertNotNull(clusterSettingsHandler);
                 assertThat(clusterSettingsHandler.keys(), containsInAnyOrder("setting_1", "setting_2"));
                 ReservedStateHandlerMetadata projectSettingsHandler = handlers.get("project_settings");
-                assertNotNull(projectSettingsHandler);
-                assertThat(projectSettingsHandler.keys(), containsInAnyOrder("setting_1"));
+                if (multiprojectEnabled) {
+                    assertNotNull(projectSettingsHandler);
+                    assertThat(projectSettingsHandler.keys(), containsInAnyOrder("setting_1"));
+                } else {
+                    assertNull(projectSettingsHandler);
+                }
             }
         } else {
             assertThat(metadata.getProject(projectId).indices(), anEmptyMap());

